File: Programmers/python/[1차]비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    
    for i in range(n):
        line = ""
        
        binary1 = bin(arr1[i])[2:].zfill(n)
        binary2 = bin(arr2[i])[2:].zfill(n)
        
        map_len = len(binary1)
        # print(map_len)
        for k in range(map_len):
            target1 = int(binary1[k])
            target2 = int(binary2[k])
            
            # print("{}, {}".format(target1, target2))
            
            # 둘 중 하나라도 1(벽)이라면
            if target1 or target2:
                # print("벽")
                line += "#"
            else:
                # print("공간")
                line += " "
        answer.append(line)
        # print()
    
    return answer

File: Programmers/java/[1차]비밀지도.java
import java.util.*;

class Solution {
    public List<String> solution(int n, int[] arr1, int[] arr2) {
        List<String> answer = new ArrayList<>();
        
        for(int i = 0; i < arr1.length; i++) {
            int num1 = arr1[i];
            int num2 = arr2[i];
            
            String converted1 = String.format("%" + n + "s", Long.toBinaryString(num1).replace(' ', '0'));
            String converted2 = String.format("%" + n + "s", Long.toBinaryString(num2).replace(' ', '0'));
            
            answer.add(getConverted(converted1, converted2));
        }
    
        return answer;
    }
    
    String getConverted(String converted1, String converted2) {
        StringBuilder sb = new StringBuilder();
        
        for(int i = 0; i < converted1.length(); i++) {
            if(converted1.charAt(i) == '1' || converted2.charAt(i) == '1') {
                sb.append('#');
                continue;
            }
            sb.append(' ');
        }
        return sb.toString();
    }
}
